devicereg/devicereg.proto

```diff
@@ -97,13 +136,206 @@ message RevokeDeviceRequest {
+  // defined for the specified sensor type. The default value is `SHARE` which
+  // implies sharing the data at full resolution. If this type is specified, it
+  // is an error if either of `buckets` or `interval` is also supplied.
+  enum Action {
+    SHARE = 0;
+    BIN = 1;
+    MOVING_AVG = 2;
+  }
+
+  // The specific action this entitlement defines for the sensor type. This is a
+  // required field.
+  Action action = 2;
+
+  // The bins attribute is used to specify the the bins into which incoming
+  // values should be classified. Each element in the list is the upper
+  // inclusive bound of a bin. The values submitted must be sorted in strictly
+  // increasing order. There is no need to add a highest bin with +Inf bound, it
+  // will be added implicitly. This field is optional unless an Action of `BIN`
+  // has been requested, in which case it is required. It is an error to send
+  // values for this attribute unless the value of Action is `BIN`.
+  repeated double bins = 3;
+
+  // This attribute is used to control the entitlement in the case for which we
+  // have specified an action type representing a moving average. It represents
+  // the interval in seconds over which the moving average should be calculated,
+  // e.g. for a 15 minute moving average the value supplied here would be 900.
+  // This field is optional unless an Action of `MOVING_AVG` has been specified,
+  // in which case it is required. It is an error to send a value for this
+  // attribute unless the value of Action is `MOVING_AVG`.
+  uint32 interval = 4;
+}
+
+// CreateEntitlementPolicyRequest is a message sent to the device registration
+// service to create a new entitlement policy. An entitlement policy is a
+// collection of one or more "Entitlements". A single Entitlement specifies an
+// operation to be performed on a single data channel being published by a
+// SmartCitizen device. The policy as a whole is comprised of one or more
+// Entitlements.
+message CreateEntitlementPolicyRequest {
+  // The list of entitlements we wish to create for the policy.
+  repeated Entitlement entitlements = 1;
+}
+
+// CreateEntitlementPolicyResponse is a message returned by the service after a
+// policy has been created. On creating a policy the device registration service
+// also creates a key pair for the policy. The public part of this key pair will
+// be used by the encoder in order to encrypt data for the specified recipient.
+// In order to decrypt data at the other end, the private key must be
+// distributed to the recipient, and they will then be able to use this key to
+// decrypt the data. It is out of scope for this service to address how this key
+// is distributed to the correct recipient. The returned message below contains
+// a unique identifier of the policy allocated by the service, as well as the
+// newly created key pair.
+message CreateEntitlementPolicyResponse {
```

Comment:
If data produced by a sensor is encrypted after a policy has been defined - how does the owner of the device read the data she owns? Does she need to create a new policy for herself?
I know that in reality data will be available in the open on the Smart Citizen platform but it still raises the (perhaps slightly out of scope) question.

Reply:
When a device is claimed we currently set up a full resolution stream with their credentials meaning they can retrieve from the store.

Reply:
When a device is claimed we currently set up a full resolution stream with their credentials meaning they can retrieve from the store.

Reply:
👍makes sense, I wasn't sure if this was the case